pkg/config/draftconfig.go

```diff
@@ -68,28 +69,60 @@ func (d *DraftConfig) GetNameOverride(path string) string {
 	return prefix
 }
 
-// ApplyDefaultVariables will apply the defaults to variables that are not already set
-func (d *DraftConfig) ApplyDefaultVariables(customInputs map[string]string) error {
-	varIdxMap := VariableIdxMap(d.Variables)
+func (d *DraftConfig) AddVariable(name, value string) {
+	d.Variables = append(d.Variables, &BuilderVar{
+		Name:  name,
+		Value: value,
+	})
+
+	d.varIdxMap[name] = len(d.Variables) - 1
+}
+
+func initVarIdxMap(variables []*BuilderVar) map[string]int {
+	varIdxMap := make(map[string]int)
+
+	for i, variable := range variables {
+		varIdxMap[variable.Name] = i
+	}
+
+	return varIdxMap
+}
 
+func (d *DraftConfig) GetVariable(name string) (*BuilderVar, error) {
+	if d.varIdxMap == nil {
+		d.varIdxMap = initVarIdxMap(d.Variables)
+	}
+
+	if idx, ok := d.varIdxMap[name]; ok {
```

Comment:
this method only update the first time we call it here.
it would be more likely to remain in sync if it scanned for a missing variable and then updated its entry in the map to avoid state drift since the variables slice is mutable